backend/internal/application/usecase/hello_world_usecase_test.go

```diff
@@ -0,0 +1,119 @@
+package usecase
```

Comment:
package名、`usecase_test`にしたら別のパッケージとして認識されて（Goでは同一フォルダのpackageは原則同じにしないといけないけど、これだけ許容される）、プライベートな関数がテストが見えなくなるからブラックボックス的なテストにできる！

ref: https://qiita.com/interharuki/items/18f31546947608398ebd

```suggestion
package usecase_test
```

Reply:
usecase_testとやるとMissingLitField: unknown field r in struct literal of type usecase.HelloWorldcaseって出てきて、HelloWorldcase見てみると構造体rなのよね、、、

```go
func TestNewHelloWorldUsecase(t *testing.T) {
	type args struct {
		r repository.HelloWorldRepository
	}
	tests := []struct {
		name string
		args args
		want usecase.HelloWorldUsecase
	}{
		{
			name: "success",
			args: args{
				r: &repository_mock.MockHelloWorldRepository{},
			},
			want: usecase.HelloWorldUsecase{
				r: &repository_mock.MockHelloWorldRepository{}, // ここでエラーになる
			},
		},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			u := usecase.NewHelloWorldUsecase(tt.args.r)
			assert.Equal(t, u, tt.want)
		})
	}
}
```

Reply:
ここは解決済み